logoapp/js/dot_data.js

```diff
@@ -0,0 +1,1028 @@
+	this.isDrawHover=false;
+	this.isPressMain = false;
+
+	this.helpButton = document.getElementById("helpButton");
+	this.helpButton.addEventListener('click', this.onClickHelpButton.bind(this), false);
+	this.undoButton = document.getElementById("undoButton");
+	this.undoButton.addEventListener('click', this.onClickUndoButton.bind(this), false);
+	this.clearButton = document.getElementById("clearButton");
+	this.clearButton.addEventListener('click', this.onClickClearButton.bind(this), false);
+	this.copyButton = document.getElementById("copyButton");
+	this.copyButton.addEventListener('click', this.onClickCopyButton.bind(this), false);
+	this.playButton = document.getElementById("playButton");
+	this.playButton.addEventListener('click', this.onClickPlayButton.bind(this), false);
+	this.completeButton = document.getElementById("completeButton");
+	this.completeButton.addEventListener('click', this.onClickCompleteButton.bind(this), false);
+	this.notCompleteButton = document.getElementById("notCompleteButton");
+	this.notCompleteButton.addEventListener('click', this.onClickNotCompleteButton.bind(this), false);
+	this.uploadButton = document.getElementById("uploadButton");
+	this.uploadButton.addEventListener('click', this.onClickUploadButton.bind(this), false);
+};
+DotAppFramework.prototype.start = function () {
+	setInterval(this.loop.bind(this) , 33 );
+	this.drawButton();
+};
+DotAppFramework.prototype.copyArrDotFrame = function(arrDotFrame) {
+	var newArray = new Array(8);
+		for(var i=0; i<newArray.length; i++) {
+			newArray[i] = arrDotFrame[i].copy();
+	}
+	return newArray;
+};
+DotAppFramework.prototype.unshiftHistoryArrDotFrame = function() {
+	this.historyArrDotFrame.unshift(this.copyArrDotFrame(this.arrDotFrame));
+};
+DotAppFramework.prototype.onClickMain = function(e) {
+	var rect = e.target.getBoundingClientRect();
+	var x = e.clientX - rect.left;
+	var y = e.clientY - rect.top;
+	this.mainView.mousePos = new Vector2(x, y);
+};
+DotAppFramework.prototype.onMouseDownMain = function() {
+	this.isPressMain = true;
+
+	if(this.isDraw) {
+		if(this.isHoverDot) {
+			if(this.hoverDotNum==this.preDrawNum) {
+				if(this.arrDotFrame[this.currentFrame-1].arrDot[this.hoverDotNum].isDraw) {
+					this.arrDotFrame[this.currentFrame-1].arrDot[this.hoverDotNum].isDraw = false;
+				}
+				else {
+					this.arrDotFrame[this.currentFrame-1].arrDot[this.hoverDotNum].isDraw = true;
+					this.arrDotFrame[this.currentFrame-1].arrDot[this.hoverDotNum].color = this.currentColor;
+				}
+			}
+			this.addHistory();
+			this.buttonActive.clear=false;
+			for(var i=0; i<DOT_NUM; i++) {
+				if(this.arrDotFrame[this.currentFrame-1].arrDot[i].isDraw) {
+					this.buttonActive.clear=true;
+					break;
+				}
+			}
+		}
+		this.isDrawHover=true;
+	}
+}
+DotAppFramework.prototype.onMouseUpMain = function() {
+	this.isPressMain = false;
+	this.isDrawHover = false;
+}
+DotAppFramework.prototype.addHistory = function() {
+	if(this.historyPos!= 0) {

```

Comment:
__[jshint]__
*  Use '!==' to compare with '0'.


<!-- Comment by [SideCI](https://sideci.com) -->